examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -0,0 +1,431 @@
+
+        elif handle_type == nonbonded.LennardJonesHandler:
+            # sanity check again, even though we don't have other lj methods currently
+            assert handle_type == type(forcefield.lj_handle)
+
+
+def _print_result(lamb, result):
+    """
+    TODO: include type hint here for ambiguous argument name "result"
+    TODO: add units
+    TODO: move message into a __str__ method for a result object?
+    TODO: replace print with logger
+    """
+    bonded_du_dl, nonbonded_du_dl, grads_and_handles = result
+
+    unit = "kJ/mol"
+
+    message = f"""
+    lambda {lamb:.3f}
+    bonded dU/dlambda (in {unit})
+        mean = {np.mean(bonded_du_dl):.3f}
+        stddev = {np.std(bonded_du_dl):.3f}
+    nonbonded dU/dlambda (in {unit})
+        mean = {np.mean(nonbonded_du_dl):.3f}
+        stddev = {np.std(nonbonded_du_dl):.3f}
+    """
+    print(message)
+
+
+def _mean_du_dlambda(result):
+    """summarize result of rfe.host_edge into mean du/dl
+
+    TODO: refactor where this analysis step occurs
+    """
+    bonded_du_dl, nonbonded_du_dl, _ = result
+    return np.mean(bonded_du_dl + nonbonded_du_dl)
+
+
+def _update_combined_handle_and_grads(ghs, combined_handle_and_grads):
+    # use gradient information from the endpoints
+    for (grad_lhs, handle_type_lhs), (grad_rhs, handle_type_rhs) in zip(ghs[0], ghs[-1]):
+        assert handle_type_lhs == handle_type_rhs  # ffs are forked so the return handler isn't same object as that of ff
+        grad = grad_rhs - grad_lhs
+
+        # complex - solvent
+        if handle_type_lhs not in combined_handle_and_grads:
+            combined_handle_and_grads[handle_type_lhs] = grad
+        else:
+            combined_handle_and_grads[handle_type_lhs] -= grad
+
+
+class ThermodynamicIntegrationResult:
+    def __init__(self, lambda_schedule, results):
+        self.lambda_schedule = lambda_schedule
+        self.results = results
+
+    @staticmethod
+    def _save_result(name, lamb, result, rfe, configuration):
+        """
+        TODO: more compact way to store the relevant information about which transformation was computed
+        TODO: include type hint here for ambiguous argument name "result"
+        TODO: add units
+        TODO: move this into part of a .save() method for a RelativeFreeEnergy result object?
+        TODO: use state index i rather than lambda in filename
+        TODO: include also dU/dparams here?
+        """
+        # unpack result tuple
+        bonded_du_dl, nonbonded_du_dl, grads_and_handles = result
+
+        # save du/dlambda trajectories
+        du_dl_filename = path_to_results.joinpath(
+            f"du_dlambda trajectories ({name}, lambda={lamb:.3f})")
+        print(f'saving bonded and nonbonded du/dlambda trajectories to {du_dl_filename}...')
+        np.savez(du_dl_filename, bonded_du_dl=bonded_du_dl, nonbonded_du_dl=nonbonded_du_dl, rfe=rfe,
+                 configuration=configuration)
```

Comment:
kudos for saving the configuration!

Reply:
Saving the `rfe` object in this archive was definitely redundant here, as I ended up storing `2 * n_lambdas` copies of a serialized `rfe` per TI calculation... (These were also stored in a way that required `np.load(..., allow_pickle=True)`, which isn't ideal.)

Since we have already stored the serialized `rfe`'s in an ordered list in the same directory, it suffices to store the integer index of the `rfe` used in each optimizer step, addressed here: https://github.com/proteneer/timemachine/pull/331/commits/666e4cd91808cbfae314bb9ac4262772808b1ef2